vendor/knative.dev/pkg/test/spoof/openshift_error.go

```diff
@@ -0,0 +1,44 @@
+package spoof
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+// OpenShiftErrorRetryChecker implements the ErrorRetryChecker for retrying on OpenShift error.
+func OpenShiftErrorRetryChecker(err error) (bool, error) {
+	if isUnknownAuthority(err) {
```

Comment:
I feel like we should patch this into the default checker, since that's used in at least one spot upstream on it's own, see https://github.com/knative/serving/blob/68dba4ae68ca817f168e4582f6fb9e0c1acaaa62/test/conformance/runtime/readiness_probe_test.go#L187